### PR TITLE
[DA-797] fix sample status time sql error and relevant test cases

### PR DESCRIPTION
**Pull request by wangy70**
change relevant test cases to force to use mysql and fix the issues cause by this change.

### CODE REVIEW

File: rest-api/dao/participant_dao.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def get_id(self, obj):
   def insert_with_session(self, session, obj):
     obj.hpoId = self._get_hpo_id(obj)
     obj.version = 1
-    obj.signUpTime = clock.CLOCK.now()
+    obj.signUpTime = clock.CLOCK.now().replace(microsecond=0)
```
Comment (wangy70) [inline rest-api/dao/participant_dao.py]: MySQL doesn't save microsecond and will automatically round to second.

File: rest-api/dao/participant_summary_dao.py [hunk 2]
```diff
@@ -165,35 +165,33 @@ def _get_sample_status_time_sql_and_params():
                                             .lower() for item in baseline_ppi_module_fields])
 
   sub_sql = """
-  (
     SELECT 
-      MAX(
+      participant_id,
+      GREATEST(
         consent_for_electronic_health_records_time,
         physical_measurements_finalized_time,
```
Comment (wangy70) [inline rest-api/dao/participant_summary_dao.py]: Change the sql to MySQL grammar.

File: rest-api/model/questionnaire.py [hunk 3]
```diff
@@ -80,7 +80,7 @@ class QuestionnaireQuestion(Base):
       'questionnaire_question_id', Integer, primary_key=True)
   questionnaireId = Column('questionnaire_id', Integer)
   questionnaireVersion = Column('questionnaire_version', Integer)
-  linkId = Column('link_id', String(20))
+  linkId = Column('link_id', String(40))
   codeId = Column(
```
Comment (wangy70) [inline rest-api/model/questionnaire.py]: In our test data, we have some linkId length more than 20, change the column to 40.

File: rest-api/test/unit_test/api_test/participant_summary_api_test.py [hunk 4]
```diff
@@ -94,6 +94,9 @@ class ParticipantSummaryApiTest(FlaskTestBase):
       'firstName', 'middleName', 'lastName', 'streetAddress', 'city', 'phoneNumber', 'zipCode'
   )
 
+  def setUp(self):
+    super(ParticipantSummaryApiTest, self).setUp(use_mysql=True)
+
```
Comment (wangy70) [inline rest-api/test/unit_test/api_test/participant_summary_api_test.py]: force this test case to use MySQL

File: rest-api/test/unit_test/dao_test/participant_summary_dao_test.py [hunk 5]
```diff
@@ -21,7 +21,7 @@
 
 class ParticipantSummaryDaoTest(NdbTestBase):
   def setUp(self):
-    super(ParticipantSummaryDaoTest, self).setUp()
+    super(ParticipantSummaryDaoTest, self).setUp(use_mysql=True)
     self.dao = ParticipantSummaryDao()
```
Comment (wangy70) [inline rest-api/test/unit_test/dao_test/participant_summary_dao_test.py]: force this test case to use MySQL

File: rest-api/test/unit_test/dao_test/participant_summary_dao_test.py [hunk 6]
```diff
@@ -254,10 +256,13 @@ def add_sample(participant, test_code, sample_id):
     M_baseline_samples = self._insert(Participant(participantId=9, biobankId=99))
     add_sample(M_baseline_samples, baseline_tests[0], '999')
     M_first_update = self.dao.get(M_baseline_samples.participantId)
-
+    # sleep 1 sec to make lastModified different
+    time.sleep(1)
     self.dao.update_from_biobank_stored_samples()
```
Comment (wangy70) [inline rest-api/test/unit_test/dao_test/participant_summary_dao_test.py]: the precision of MySQL only to second, so we need to sleep at least 1 sec to make the lastModify timestamp change.

File: rest-api/test/unit_test/offline_test/biobank_samples_pipeline_test.py [hunk 7]
```diff
@@ -25,7 +25,7 @@
 
 class BiobankSamplesPipelineTest(CloudStorageSqlTestBase, NdbTestBase):
   def setUp(self):
-    super(BiobankSamplesPipelineTest, self).setUp()
+    super(BiobankSamplesPipelineTest, self).setUp(use_mysql=True)
     NdbTestBase.doSetUp(self)
```
Comment (wangy70) [inline rest-api/test/unit_test/offline_test/biobank_samples_pipeline_test.py]: force this test case to use MySQL

File: rest-api/alembic/versions/c06896a4fb4d_change_questionnaire_linkid_length_to_.py [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    globals()["upgrade_%s" % engine_name]()
+
+
+def downgrade(engine_name):
+    globals()["downgrade_%s" % engine_name]()
+
+
+
+def upgrade_rdr():
+    # ### commands auto generated by Alembic - please adjust! ###
```
Comment (Seeker1911) [inline rest-api/alembic/versions/c06896a4fb4d_change_questionnaire_linkid_length_to_.py]: Looks like the change didn't apply.

File: rest-api/alembic/versions/041fdb188c55_change_linkid_length_to_40.py [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    globals()["downgrade_%s" % engine_name]()
+
+
+
+def upgrade_rdr():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.alter_column('questionnaire_question', 'link_id',
+               existing_type=mysql.VARCHAR(length=20),
+               type_=sa.String(length=40),
+               existing_nullable=True)
```
Comment (wangy70) [inline rest-api/alembic/versions/041fdb188c55_change_linkid_length_to_40.py]: New alembic file uploaded.

**Pull request merged**